build/source/dshare/outpt_stat.f90

```diff
@@ -0,0 +1,294 @@
+    case (iLookStat%maxi)                          ! maximum over period
+     tstat(iStat) = -huge(tstat(iStat))            ! resets stat at beginning of period
+    case (iLookStat%mode)                          ! mode over period (does not work)
+     tstat(iStat) = -9999.
+   endselect
+  enddo ! iStat 
+ endif
+
+ ! ---------------------------------------------
+ ! Calculate each statistic that is requested by user
+ ! ---------------------------------------------
+ do iStat = 1,maxVarStat                           ! loop through output statistics
+  if (.not.meta%statFlag(iStat)) cycle             ! do not bother if output flag is off
+  if (meta%varType.ne.iLookVarType%scalarv) cycle  ! only calculate stats for scalars 
+  select case(iStat)                               ! act depending on the statistic 
+   case (iLookStat%totl)                           ! summation over period
+    tstat(iStat) = tstat(iStat) + tdata            ! into summation
+   case (iLookStat%inst)                           ! instantaneous
+    tstat(iStat) = tdata                                        
+   case (iLookStat%mean)                           ! mean over period
+    tstat(iStat) = tstat(iStat) + tdata            ! adds timestep to sum 
+   case (iLookStat%vari)                           ! variance over period
+    tstat(iStat) = tstat(iStat) + tdata**2         ! sum into E[X^2] term
+    tstat(maxVarStat+1) = tstat(maxVarStat+1) + tdata  ! sum into E[X]^2 term        
+   case (iLookStat%mini)                           ! minimum over period
+    if (tdata.le.tstat(iStat)) tstat(iStat) = tdata! overwrites minimum iff 
+   case (iLookStat%maxi)                           ! maximum over period
+    if (tdata.ge.tstat(iStat)) tstat(iStat) = tdata! overwrites maximum iff 
+   case (iLookStat%mode)                           ! (does not work)
+    tstat(iStat) = -9999. 
+  endselect
+ enddo ! iStat 
+
+ ! ---------------------------------------------
+ ! finalize statistics at end of frequenncy period 
+ ! ---------------------------------------------
+ if (mod(iStep,outFreq(iFreq)).eq.0) then
+  do iStat = 1,maxVarStat                          ! loop through output statistics
+   if (.not.meta%statFlag(iStat)) cycle            ! do not bother if output flag is off
+   if (meta%vartype.ne.iLookVarType%scalarv) cycle ! only calculate stats for scalars 
+   select case(iStat)                              ! act depending on the statistic 
+    case (iLookStat%mean)                          ! mean over period
+     tstat(iStat) = tstat(iStat)/outFreq(iFreq)    ! normalize sum into mean
+    case (iLookStat%vari)                          ! variance over period
+     tstat(maxVarStat+1) = tstat(maxVarStat+1)/outFreq(iFreq) ! E[X] term
+     tstat(iStat) = tstat(iStat)/outFreq(iFreq) - tstat(maxVarStat+1)**2 ! full variance
+   endselect
+  enddo ! iStat 
+ endif
+
+ ! pack back into struc
+ selecttype (stat)
+  typeis (ilength); stat%dat = int(tstat)
```

Comment:
It's a good point. I changed it according to your suggestion so that all stats are now dp's. The reason that I had it this way was that the index variables were the only integer stats, and they could only be output at the model timestep (iLookStat%inst). But this is non-general, so it is better to output them as reals. Fixed.